### ISSUE TITLE
Heater simulation

### ISSUE THREAD

**Issue by matscb** [bwp91/homebridge-ewelink#309]
Heater simulation is not recognized in HomeDash app

Thanks for a really nice plug-in. Works really good and is easy to use and setup. 

I use Sonoff TH16 to control my electrical heaters and your plugin to get them into HomeKit using the Heater simulation. Works and I can see the heater in most HomeKit apps like in the Apple Home app.  

I use the HomeDash app to monitor my system and in this app the Heater simulation does not show correctly although other similar devices works (like the Tado heat pump/AC control). It is probably something wrong at the HomeDash end but as the support for this app is very limited I took chance to check with you if this is something you heard about and been looking at?

Another question. The heater simulation in apps like Home+ and Eve shows an attribute  ”heater cooles state” that seems to be able to get three values, automatic, cooling and heating but only seems to allow automatic. E.g the Eve app gets an error if trying to set to other than heating. Is this the way a HomeKit Heather normally works?

As said, probably not your issue but I took a chance ….

Mats Andersson
Sweden

**Comment by bwp91** [comment]
Hi @matscb 

The `HeaterCooler` HomeKit accessory allows for the `Active` characteristic to be ON or OFF and for the `TargetHeaterCooler` state to be AUTO, HEAT and COOL.

Since for this simulation there is no use for the HEAT and COOL target states, I have disabled them from appearing in the Home app. So the Home app just shows an ON/OFF button rather than a list of the unsupported modes. I believe this makes it clearer how to use it.

With regards to the Eve app, I thought that the HEAT and COOL modes would be hidden too since the plugin doesn't allow for these states. I remember testing this a while ago and these modes were hidden. Perhaps something has changed and they are visible again - I will have to find my TH16 device from the cupboard somewhere to check this out! I have never used the Home+ app.

I have never heard of the HomeDash app, though it is likely not showing the option for the `Active` characteristic to be able to turn it on and off? Could you post a screenshot of how it looks?

I primarily design all the different homekit accessory types to look best in the Home app, and try to strip away features that are unnecessary so it looks and works as well as it can but also as simple as it can look.

**Comment by matscb** [comment]
Hi Ben,

Once again, nice Ewelink plugin!

Homedash is an app used to create visually appealing HomeKit dashboards. I use an ”old ”iPad” as control center on the wall. I think it is really nice but I have noticed that there has been no updates on the app for quite a while. I tried to contact their support link but so far no result …???

I think there a limited free version if you want to try it out?

Seems that the Heater is shown on in what is called the ”control” section of the Homedash app (see hardcopy 1) but does not recognise in the when using the Heating Control Widget used when creating the dashboards (hardcopy 2). In my understanding the room with a supported Heater should be available but it is not. 

”Similar devices” e.g. Tado AC controller is shown in the widget but I imagine this another accessory type (see hardcopy 3).

I thank you for trying to help out here but I imagine the problem is probably in Homedash. Hopefully they will come back to me.

Rgds,
Mats
